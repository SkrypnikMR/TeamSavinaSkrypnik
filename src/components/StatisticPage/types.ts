import { TFullStat } from '../../store/statistic/types';

export type TStatisticPage = {
    userLogin: string;
    getFullStat: () => void;
    fullStatistic: TFullStat;
}
