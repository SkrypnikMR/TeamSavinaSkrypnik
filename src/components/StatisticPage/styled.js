import styled from 'styled-components';

export const StStatisticPage = styled.div`
  min-height: 100vh;
  background: #404040;
  display: flex;
  flex-direction: column;
  .timeless{
      background: #696969;
      height: 84vh;
      margin: 10px;
      color: white;
      font-size: 25px;
      text-align: center;
  }
`;