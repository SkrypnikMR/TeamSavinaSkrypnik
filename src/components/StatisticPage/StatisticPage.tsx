import React, { useEffect } from 'react';
import Header from '../Header';
import Footer from '../Footer';
import StatisticCard from '../StatisticCard';
import { StStatisticPage, StStatisticContent } from './styled';
import { TStatGame } from '../../store/statistic/types';
import { TStatisticPage } from './types';

const StatisticPage = ({ getFullStat, fullStatistic, userLogin }: TStatisticPage) => {
    useEffect(() => getFullStat(), []);
    return (
        <StStatisticPage>
            <Header/>
            <StStatisticContent>
{
                fullStatistic.length > 0
                        ? fullStatistic.map((game: TStatGame) => (
                            <StatisticCard
                                key={game.uuidGame}
                                draw={game.draw}
                                winnerLogin={game.winnerLogin}
                                gameType={game.gameType}
                                userLogin={userLogin}
                                creatorLogin={game.creatorLogin}
                                guestLogin={game.guestLogin}
                                startTime={game.startTime}
                                finishTime={game.finishTime}
                            />
                    ))
                    : 'nothing to see'
            }
            </StStatisticContent>
            <Footer/>
        </StStatisticPage>
    );
};

export default StatisticPage;
