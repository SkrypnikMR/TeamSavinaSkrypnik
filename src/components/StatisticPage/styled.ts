import styled from 'styled-components';

import { darkGrey, midGrey, white } from '../UI/baseLayout';

export const StStatisticPage = styled.div`
  min-height: 100vh;
  background: ${darkGrey};
  display: flex;
  flex-direction: column;
`;
export const StStatisticContent = styled.div`
  background: ${midGrey};
  border-radius: 3px;
  height: 82vh;
  margin: 10px;
  padding: 10px;
  color: ${white};
  display: grid;
  grid-template-columns: repeat(5,1fr);
  justify-items: center;
  align-items: center;
  grid-gap: 10px;
      overflow-y: auto;
      overflow-x: hidden;
        ::-webkit-scrollbar {
          width: 8px;
          height: 8px;
        }
            ::-webkit-scrollbar-thumb {
              width: 8px;
              background-color: ${({ color = white }) => color};
            }
`;
