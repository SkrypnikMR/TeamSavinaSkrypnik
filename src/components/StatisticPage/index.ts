import { connect } from 'react-redux';
import { Dispatch } from 'redux';
import { getFullStat } from '../../store/statistic/actions';
import StatisticPage from './StatisticPage';
import { fullStatSelector } from '../../store/statistic/selectors';
import { ApplicationState } from '../../store/types';

const mapStateToProps = (state: ApplicationState) => ({ fullStatistic: fullStatSelector(state) });

const mapDispatchToProps = (dispatch: Dispatch) => ({
    getFullStat: () => dispatch(getFullStat()),
});

export default connect(mapStateToProps, mapDispatchToProps)(StatisticPage);
