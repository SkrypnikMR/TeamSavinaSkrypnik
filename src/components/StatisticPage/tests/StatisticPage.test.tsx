import React from 'react';
import configureStore from 'redux-mock-store';
import StatisticPage from '../StatisticPage';
import { shallowSmart, mountSmart } from '../../../../__tests__/testHelper';

jest.mock('../../../index', () => ({ store: {} }));

const mockStore = configureStore();
const store = mockStore({
  game: {
        rooms: [],
        fullStatistic: [],
  },
});

describe('StatisticPage', () => {
    const props = {
        fullStatistic: [],
        userLogin: 'userLogin',
        getFullStat: jest.fn(),
    };
    it('Should match snapshot', () => {
        const component = shallowSmart(<StatisticPage {...props} />, store);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render StSingleRoom', () => {
        const component = mountSmart(<StatisticPage {...props}/>, store);
        expect(component.find('styled__StStatisticPage')).toHaveLength(1);
    });
    it('should render Header', () => {
        const component = mountSmart(<StatisticPage {...props}/>, store);
        expect(component.find('Header')).toHaveLength(1);
    });
    it('should render Footer', () => {
        const component = mountSmart(<StatisticPage {...props}/>, store);
        expect(component.find('Footer')).toHaveLength(1);
    });
    it('should render StatisticCard', () => {
        props.fullStatistic = [
            {
                uuidGame: '21312321321321',
                draw: false,
                winnerLogin: 'userLogin',
                gameType: 'Chekers',
                userLogin: 'userLogin',
                creatorLogin: 'userLogin',
                guestLogin: 'Bot',
                startTime: 123213213213213,
                finishTime: 12312312321321,
            },
        ];
        const component = mountSmart(<StatisticPage {...props}/>, store);
        expect(component.find('StatisticCard')).toHaveLength(1);
    });
});
