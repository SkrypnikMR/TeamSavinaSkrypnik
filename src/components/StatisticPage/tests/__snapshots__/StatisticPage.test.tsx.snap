// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatisticPage Should match snapshot 1`] = `"<div class=\\"styled__StStatisticPage-sc-f8n9su-0 xKhCj\\"><div class=\\"styled__StHeader-sc-1d7yllk-0 ibTsld\\"><div class=\\"styled__StLogo-sc-1lhyjdk-0 eFYqgc\\"><img src=\\"public/assets/images/game-development.png\\"/><div><button type=\\"button\\" font-size=\\"20px\\" class=\\"styled__StButton-sc-1y35zax-0 klRECi\\">game_zone</button><button type=\\"button\\" font-size=\\"20px\\" class=\\"styled__StButton-sc-1y35zax-0 WqRKw\\">statistics</button></div></div><div class=\\"styled__StControl-sc-1l7fu2q-0 bcijKV\\"> <button id=\\"theme_btn\\" type=\\"button\\" value=\\"dark\\" color=\\"white\\" font-size=\\"26px\\" width=\\"40px\\" class=\\"styled__StButton-sc-1y35zax-0 dxIorm\\">ðŸ’¡</button><button id=\\"en\\" type=\\"button\\" value=\\"en\\" color=\\"white\\" font-size=\\"26px\\" width=\\"40px\\" class=\\"styled__StButton-sc-1y35zax-0 hfKumd\\">english_translate_btn</button><button id=\\"ru\\" type=\\"button\\" value=\\"ru\\" color=\\"white\\" font-size=\\"26px\\" width=\\"40px\\" class=\\"styled__StButton-sc-1y35zax-0 edZCPS\\">russian_translate_btn</button><img src=\\"public/assets/images/log-out.png\\"/></div></div><div class=\\"styled__StStatisticContent-sc-f8n9su-1 dTNDIj\\">nothing to see</div><div class=\\"styled__StFooter-sc-1xbiugi-0 bClKzk\\">create_forÂ <a href=\\"https://deveducation.com/\\" target=\\"_blank\\" rel=\\"noreferrer\\">dev{Education}</a></div></div>"`;
